vsphere/resource_vsphere_role.go

```diff
@@ -0,0 +1,142 @@
+package vsphere
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-provider-vsphere/vsphere/internal/helper/structure"
+	"github.com/vmware/govmomi/object"
+	"log"
+	"strconv"
+	"strings"
+)
+
+func resourceVsphereRole() *schema.Resource {
```

Comment:
how are global permissions/roles managed?

Reply:
They are not supported as of now, only entity level permissions are supported.
I couldn't find an API for global permissions.

Reply:
Is it a requirement to add support for global permissions? Maybe a good idea to check with Santosh.

Reply:
Yes, I highly suggest that the ability to add global permissions is indeed important. I've already run into instances where I need to automate this for use by vRealize Suite service accounts that need this scope of access across Cloud Foundation workload domains. For example, vRealize Automation required global permissions to use the content library based on the hierarchy.